### PR TITLE
Anomaly Armour Research

### PR DESCRIPTION
**Pull request by imtakingabreakdontatme**
:cl: Kor
add: Science may now research reactive armour shells, which create different armours when combined with different anomaly cores. They are constructable from the Research and Engineering lathes.
/:cl:

Why:  Anomaly cores have few uses, science could use something combat related as a bribe for RnD now that they dont get any guns, and I want to see engineering pushing the engine towards near meltdown to try and harvest anomalies (would obviously need a way to make those capturable though).

I dont know where to put the node in the "web" really, so suggestions are welcome. 

The reactive armour shell accepts anomaly cores to build various armours as follows

Flux - Tesla armour

Pyro - Fire armour

Gravity - Repulse armour (new in this PR)

Bluespace - Teleport Armour

Everything else - Defaults to stealth armour as a fallback

### CODE REVIEW

**Comment by PKPenguin321** [comment]
reactive teleport armor is stupid

**Comment by Bombpants** [comment]
So the tesla and fire armor shoots out fire and lightening when attacked?

**Comment by Barhandar** [comment]
Oh wow, autonomous griefing tools, don't even have to do anything.

Tesla one does actual tesla_zap too meaning some things will be unharmed and others will just get DELEET.

File: code/modules/clothing/suits/reactive_armour.dm [hunk 1]
```diff
@@ -0,0 +1,235 @@
...
+	icon_state = "reactiveoff"
+	item_state = "reactiveoff"
+	blood_overlay_type = "armor"
+	armor = list("melee" = 0, "bullet" = 0, "laser" = 0, "energy" = 0, "bomb" = 0, "bio" = 0, "rad" = 0, "fire" = 100, "acid" = 100)
+	actions_types = list(/datum/action/item_action/toggle)
+	resistance_flags = INDESTRUCTIBLE | LAVA_PROOF | FIRE_PROOF | ACID_PROOF
+	hit_reaction_chance = 50
+
+/obj/item/clothing/suit/armor/reactive/attack_self(mob/user)
+	src.active = !( src.active )
```
Comment (kevinz000) [inline code/modules/clothing/suits/reactive_armour.dm]: src.

File: code/modules/clothing/suits/reactive_armour.dm [hunk 2]
```diff
@@ -0,0 +1,235 @@
...
+/obj/item/clothing/suit/armor/reactive/teleport
+	name = "reactive teleport armor"
+	desc = "Someone separated our Research Director from his own head!"
+	var/tele_range = 6
+	var/rad_amount= 15
+	reactivearmor_cooldown_duration = 100
+
+/obj/item/clothing/suit/armor/reactive/teleport/hit_reaction(mob/living/carbon/human/owner, atom/movable/hitby, attack_text = "the attack", final_block_chance = 0, damage = 0, attack_type = MELEE_ATTACK)
+	if(!active)
+		return 0
```
Comment (kevinz000) [inline code/modules/clothing/suits/reactive_armour.dm]: muh booleans

**Comment by kevinz000** [comment]
@Barhandar #37369

**Comment by Shadowflame909** [comment]
>NO REACTIVE TABLE ARMOR.

Wow kor you really missed the chance at giving us the best armor in game.

**Comment by improvedname** [comment]
Bluespace core should give you 1% on making table armour

**Comment by improvedname** [comment]
Make tabling kevinz give a 5% chance to create a legendary table anonoly which tables everything in view and hardest one tk stop to make the best item ingame

**Comment by Repukan** [comment]
Don't they have the flying suit for their combat related needs?

Anyways add some way for this armor set to explode past a certain amount of damage to show how much of a prototype it is.

File: code/modules/clothing/suits/reactive_armour.dm [hunk 3]
```diff
@@ -0,0 +1,235 @@
...
+		var/armour_path = anomaly_armour_types[A.anomaly_type]
+		if(!armour_path)
+			armour_path = /obj/item/clothing/suit/armor/reactive/stealth //Lets not cheat the player if an anomaly type doesnt have its own armour coded
+		to_chat(user, "You insert [A] into the chest plate, and the armour gently hums to life.")
+		new armour_path(get_turf(src))
+		qdel(src)
+		qdel(A)
+
+//Reactive armor
+/obj/item/clothing/suit/armor/reactive
```
Comment (Fox-McCloud) [inline code/modules/clothing/suits/reactive_armour.dm]: Don't forget to make a subtype of this that has some special sprite/name/what have you, that is the steal object, so people can't just make a shell+use an anomaly to complete their objective (granted, the chances of that happening is rare, but the precedent has generally been you have to go for a one of a kind item).

**Comment by Kromgar** [comment]
Science cant print flying suits thats why you should play the game @Repukan

File: code/modules/clothing/suits/reactive_armour.dm (hunk 3 above)
Comment (ExcessiveUseOfCobblestone) [reply code/modules/clothing/suits/reactive_armour.dm]: If someone goes that far they should get the obj tbh.

**Comment by Dorsisdwarf** [comment]
Can we get the decoy belt but as a vest for this? I love the decoy belt

File: code/modules/clothing/suits/reactive_armour.dm (hunk 3 above)
Comment (KorPhaeron) [reply code/modules/clothing/suits/reactive_armour.dm]: Getting a bluespace anomaly is rare enough that I dont think a subtype is warranted

**Comment by ExcessiveUseOfCobblestone** [comment]
@Kromgar don't encourage evading lol

**Comment by KorPhaeron** [comment]
@tgstation/commit-access please review or put out of its misery

**Pull request merged**